### PR TITLE
Refactor/network module

### PR DESCRIPTION
**Pull request by KritikHub**

### CODE REVIEW

**Comment by KritikHub** [comment]
@JustManish Please review the latest changes in the branch.

File: SwiftUiMovieDb/Network/APIKey.swift [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+//
+//  Created by mac on 05/11/23.
+//
+
+import Foundation
+
+struct APIKey {
+    
+    var apiKey: String {
+        return "3492e734e03e243d1f8b59388a7b1bb2"
```
Comment (JustManish) [inline SwiftUiMovieDb/Network/APIKey.swift:13]: let's store this in xcconfig file and read from that.

File: SwiftUiMovieDb/Network/APIService.swift [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+//
+//  Created by mac on 04/11/23.
+//
+
+import Foundation
+
+class APIService {
+    
+    func makeRequest<T: URN>(with urnData: T,
+                             headers: [String: String]? = nil,
```
Comment (JustManish) [inline SwiftUiMovieDb/Network/APIService.swift:13]: why headers argument, it's not needed can get from URN itself.

File: SwiftUiMovieDb/Network/APIService.swift [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        request.httpBody = urnData.body
+        
+        let task = URLSession.shared.dataTask(with: request) { [weak self] (data, response, error) in
+            
+            guard let _self = self else {
+                completionHandler(.failure(.invalidSelf))
+                return
+            }
+            
+            if let error = error {
```
Comment (JustManish) [inline SwiftUiMovieDb/Network/APIService.swift:37]: guard will be more suitable here.

File: SwiftUiMovieDb/Protocol/DescriptiveErrorType.swift [hunk 4]
```diff
@@ -0,0 +1,79 @@
+//
+//  DescriptiveErrorType.swift
+//  SwiftUiMovieDb
+//
+//  Created by mac on 04/11/23.
+//
+
+import Foundation
+
+struct MDBErrorConstants {
```
Comment (JustManish) [inline SwiftUiMovieDb/Protocol/DescriptiveErrorType.swift:10]: This struct is not needed let's create localization file and move all the strings in it and directly use it in below enum.

File: SwiftUiMovieDb/Views/MovieBackdropCard.swift [hunk 5]
```diff
@@ -11,27 +11,37 @@ import Kingfisher
...
+            .cornerRadius(cornerRadius)
+            .shadow(radius: shadowRadius)
             Text(movie.title)
         }
-        .lineLimit(1)
+        .lineLimit(lineLimit)
     }
 }
 
+@ViewBuilder
```
Comment (JustManish) [inline SwiftUiMovieDb/Views/MovieBackdropCard.swift:39]: do we really need to annotate it with @Viewbuilder ?

File: SwiftUiMovieDb/Views/MovieBackdropCard.swift [hunk 6]
```diff
@@ -11,27 +11,37 @@ import Kingfisher
...
+func backdropPlaceholderImage() -> some View {
+    Image("PosterImagePlaceholder")
```
Comment (JustManish) [inline SwiftUiMovieDb/Views/MovieBackdropCard.swift:41]: Let's create an enum to manage all the icon or image asset to be typesafe and less errorprone, avoid using String constant's.

File: SwiftUiMovieDb/Views/MoviePosterCard.swift [hunk 7]
```diff
@@ -11,31 +11,33 @@ import Kingfisher
...
+                    .shadow(radius: imageShadowRadius)            
         }
-        .frame(width: 204, height: 306)
+        .frame(width: frameWidth, height: frameHeight)
     }
 }
 
+@ViewBuilder
+private func posterPlaceholderImage() -> some View {
+    Image("PosterImagePlaceholder")
```
Comment (JustManish) [inline SwiftUiMovieDb/Views/MoviePosterCard.swift:35]: same here

File: SwiftUiMovieDb/Views/MoviePosterCard.swift [hunk 8]
```diff
@@ -11,31 +11,33 @@ import Kingfisher
...
-            }
+                    .cornerRadius(imageCornerRadius)
+                    .shadow(radius: imageShadowRadius)            
         }
-        .frame(width: 204, height: 306)
+        .frame(width: frameWidth, height: frameHeight)
     }
 }
 
+@ViewBuilder
```
Comment (JustManish) [inline SwiftUiMovieDb/Views/MoviePosterCard.swift:33]: do we really need @ViewBuilder annotation here ? think about it and let me know why not ?

**Review by JustManish — CHANGES_REQUESTED** [review]

**Pull request closed without merging**